### PR TITLE
[addins-database] Add support for application level add-ins config file.

### PR DESCRIPTION
**Pull request by lukaszunity**
Allows applications to disable shipped add-ins by default.

### CODE REVIEW

File: Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs [hunk 1]
```diff
@@ -121,6 +117,34 @@ public AddinStatus[] GetPendingUninstalls ()
...
+		}
+		
+		public static DatabaseConfiguration ReadAppConfig()
+		{
+			var exePath = System.Reflection.Assembly.GetExecutingAssembly ().Location;
+			var exeDirectory = Path.GetDirectoryName (exePath);
+			var appAddinsConfigFilePath = Path.Combine (exeDirectory, "addins-config.xml");
+
+			if (!File.Exists (appAddinsConfigFilePath))
+				return new DatabaseConfiguration ();
```
Comment (Tak) [inline Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs:141]: Maybe do this when ReadInternal throws as well?

Comment (lukaszunity) [inline Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs:141]: As you can see in AddinDatabase.cs above, the caller checks for the existence of the file before calling DatabaseConfiguration.Read (ConfigFile) and I assume there is handling of exceptions somewhere in the call-stack. I would prefer keep the logic as it was.

File: Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs [hunk 2]
```diff
@@ -121,6 +117,34 @@ public AddinStatus[] GetPendingUninstalls ()
...
+			// Overwrite app config values with user config values
+			foreach (var entry in config.addinStatus)
+				appConfig.addinStatus [entry.Key] = entry.Value;
+
+			return appConfig;
+		}
+		
+		public static DatabaseConfiguration ReadAppConfig()
+		{
+			var exePath = System.Reflection.Assembly.GetExecutingAssembly ().Location;
```
Comment (joncham) [inline Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs]: Does this return the .exe or the assembly that the current code is in (Mono.Addins something)?

Comment (lukaszunity) [inline Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs]: I think it returns the current assembly. That's what you get when you copy-paste code :smile: I'll change the name if the assembly is returned.

Comment (lukaszunity) [inline Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs]: Fixed

**Pull request merged**